bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/service/AbstractWatchQueueReader.java

```diff
@@ -116,4 +182,27 @@ public void run() {
      *            the path of event
      */
     protected abstract void processWatchEvent(WatchEvent<?> event, WatchEvent.Kind<?> kind, Path path);
+
+    /**
+     * If the queue reader is watching the directory changes, all the watch events will be processed. Otherwise the
+     * events for directories will be skipped.
+     * 
+     * @return <code>true</code> if the directory events will be processed and <code>false</code> otherwise
+     */
+    public boolean isWatchingDirectoryChanges() {
+        return watchingDirectoryChanges;
+    }
+
+    /**
+     * If the queue reader is watching the directory changes, all the watch events will be processed. Otherwise the
+     * events for changed directories will be skipped. For example, on some platforms an event for modified directory is
+     * generated when a new file is created within the directory. However, this behavior could vary a lot, depending on
+     * the platform (for more information see "Platform dependencies" section in the {@link WatchService} documentation)
+     * 
+     * @param watchDirectoryChanges - set to <code>true</code> if the directory events have to be processed and
+     *            <code>false</code> otherwise
+     */
+    public void setWatchingDirectoryChanges(boolean watchDirectoryChanges) {
```

Comment:
Functions that are called in the constructor should be use the final keyword.
Without the final keyword a derived class could overwrite it and could stuff before the parent class is fully initialized.